src/ja/mangatoshokanz/src/eu/kanade/tachiyomi/extension/ja/mangatoshokanz/MangaToshokanZ.kt

```diff
@@ -0,0 +1,339 @@
+        val request = chain.request()
+        val response = chain.proceed(request)
+
+        // open access to R18 section
+        if (request.url.host == "r18.mangaz.com" && isR18.not()) {
+            val url = baseUrl.toHttpUrl().newBuilder()
+                .host("r18.mangaz.com")
+                .addPathSegments("attention/r18/yes")
+                .build()
+
+            val r18Request = Request.Builder()
+                .url(url)
+                .head()
+                .build()
+
+            isR18 = true
+            client.newCall(r18Request).execute()
+        }
+
+        return response
+    }
+
+    override fun popularMangaRequest(page: Int): Request {
+        val url = baseUrl.toHttpUrl().newBuilder()
+            .addEncodedPathSegments("ranking/views")
+            .build()
+
+        return GET(url, headers)
+    }
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        val document = response.asJsoup()
+        val mangas = document.selectFirst(".itemList")!!.children().mangasFromListElements()
+
+        return MangasPage(mangas, false)
+    }
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        val header = headers.newBuilder()
+            .add("X-Requested-With", "XMLHttpRequest")
+            .build()
+
+        val url = baseUrl.toHttpUrl().newBuilder()
+            .addPathSegments("title/addpage_renewal")
+            .addQueryParameter("type", "official")
+            .addQueryParameter("sort", "new")
+            .addQueryParameter("page", page.toString())
+            .build()
+
+        return GET(url, header)
+    }
+
+    override fun latestUpdatesParse(response: Response): MangasPage {
+        val document = response.asJsoup()
+        val mangas = document.selectFirst("body")!!.children().mangasFromListElements()
+
+        return MangasPage(mangas, mangas.size == LATEST_MANGA_COUNT_PER_PAGE)
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = baseUrl.toHttpUrl().newBuilder()
+            .addPathSegments("title/index")
+            .addQueryParameter("query", query)
+
+        filters.forEach { filter ->
+            when (filter) {
+                is Category -> {
+                    if (filter.state != 0) {
+                        url.addQueryParameter("category", categories[filter.state].lowercase())
+                    }
+                }
+                is Sort -> {
+                    url.addQueryParameter("sort", sortBy[filter.state].lowercase())
+                }
+                else -> {}
+            }
+        }
+
+        return GET(url.build(), headers)
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        val document = response.asJsoup()
+        val mangas = document.selectFirst(".itemList")!!.children().filter { child ->
+            child.`is`("li")
+        }.mangasFromListElements()
```

Comment:
why not ".itemList li" or ".itemList > li" ?

Reply:
there can be multiple .itemList, like in here:
https://www.mangaz.com/mens/title/index?category=mens&query=%E3%82%8F%E3%81%9F%E3%81%97&sort=new&search=input